StyleCop.Analyzers/StyleCop.Analyzers/Settings/ObjectModel/DocumentationSettings.cs

```diff
@@ -104,33 +109,12 @@ public string CopyrightText
         {
             get
             {
-                string pattern = Regex.Escape("{") + "(?<Property>[a-zA-Z0-9]+)" + Regex.Escape("}");
-                MatchEvaluator evaluator =
-                    match =>
-                    {
-                        string key = match.Groups["Property"].Value;
-                        switch (key)
-                        {
-                        case "companyName":
-                            return this.CompanyName;
-
-                        case "copyrightText":
-                            return "[CircularReference]";
+                if (this.copyrightTextCache == null)
+                {
+                    this.copyrightTextCache = this.BuildCopyrightText();
+                }
 
-                        default:
-                            string value;
-                            if (this.Variables.TryGetValue(key, out value))
-                            {
-                                return value;
-                            }
-
-                            break;
-                        }
-
-                        return "[InvalidReference]";
-                    };
-
-                return Regex.Replace(this.copyrightText, pattern, evaluator);
+                return this.copyrightTextCache;
```

Comment:
Not important, but I prefer the form:

    return this.copyright ?? (this.copyright = this.BuildCopyrightText());